src/System.Memory/src/System/ReadOnlySpan.cs

```diff
@@ -29,8 +29,6 @@ public ReadOnlySpan(T[] array)
         {
             if (array == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
-            if (default(T) == null && array.GetType() != typeof(T[]))
```

Comment:
I added netfx to the test configuration to force the tests to run against slow span.

Note that my change relies on #17388 being merged, so this PR cannot merge before #17388.